src/librustc/traits/query/dropck_outlives.rs

```diff
@@ -243,7 +243,10 @@ pub fn trivial_dropck_outlives<'tcx>(tcx: TyCtxt<'_, '_, 'tcx>, ty: Ty<'tcx>) ->
 
         ty::TyAdt(def, _) => {
             if def.is_union() {
-                // Unions never run have a dtor.
+                // Unions never have a dtor.
+                true
```

Comment:
A union implementing `Drop` itself is not a concern here?

Reply:
Good question, cc @arielb1 @pnkfelix @nikomatsakis

Reply:
Actually, yes, I think it is a concern. :D  https://github.com/rust-lang/rust/issues/52786